app/services/safe_downloader.rb

```diff
@@ -55,11 +55,18 @@ def encode_url(url)
       url = URI(url)
       raise DownloadError, 'url was invalid' unless url.respond_to?(:open)
 
+      check_url(url)
```

Comment:
I'm wondering if we could "flatten out" the exception flow here, you're introducing a new one a level deeper and it makes the code a bit less followable :thinking:

Reply:
You mean the rerise of the same type with additional info? I've kinda used the flow that is used there already.

Reply:
It's just weird that you're raising an error similarly to the previous one, just a lever deeper. But keep it so if rubocop forced you to extract this to a method :shrug:

Reply:
Oh, I see what you mean.